torch/jit/annotations.py

```diff
@@ -284,6 +284,12 @@ def try_ann_to_type(ann, loc):
     if is_dict(ann):
         key = try_ann_to_type(ann.__args__[0], loc)
         value = try_ann_to_type(ann.__args__[1], loc)
+        msg = "Unsupported annotation {} could not be resolved because {} could not be resolved."
+        # Assert if key or value is None
+        if key is None:
```

Comment:
Nit:
I think we should raise an exception with appropriate type (ValueError?) instead of asserting.

Reply:
I just followed was done with enums. I can fix this, ValueError is better